gulp/build.js

```diff
@@ -30,4 +65,65 @@ module.exports = {
 		.pipe(babel())
 		.pipe(gulp.dest(CONFIG.BUILD_DIR));
 	},
+
+	/**
+	 * This is a task that reads our `index.js` and statically analyzes it for
+	 * imports/exports. Based on that, it writes a single file for each of the
+	 * exports to the root directory. The whole purpose is to make path imports,
+	 * i.e. `import Button from 'lucid-ui/Button'`, easy.
+	 *
+	 * Because there is a potential for file system collisions, we have to
+	 * perform a check to make sure none of the files we're going to write
+	 * already exist. If they do, we'll bail out and log a waring.
+	 *
+	 * There is a sister task `jsCleanPathImports` that will remove the files
+	 * created by this task once the npm publishing is complete.
+	 */
+	jsPathImports: function() {
+		const importMap = _.fromPairs(imports());
+		const allExports = exports();
+
+		const checkFilesPromise = Promise.all(_.map(allExports, (exportName) => {
+			return new Promise((resolve, reject) => {
+				fs.stat(`${exportName}.js`, (err) => {
+					if (!err) {
+						return reject(new Error(`One of the exports (${exportName}) in index.js conflicts with a file you have on your file system (${exportName}.js). Unable to write path-import files.`));
+					}
+					return resolve();
+				});
+			});
+		}));
+
+		const writeFilesPromise = Promise.all(_.map(allExports, (exportName) => {
+			return new Promise((resolve, reject) => {
+				const exportCode = `export * from '${importMap[exportName]}';`;
+				// for some reason this doesn't pickup our .babelrc even though it's supposed to
```

Comment:
It doesn't really matter since each file is only a single line of code that's an `export`